src/components/grid/Grid.jsx

```diff
@@ -0,0 +1,74 @@
+import { useCallback, useEffect, useState } from 'react';
+
+import { useStateValue } from '../../app/state/useStateValue';
+import { constants } from '../../app/config';
+import Row from '../row/Row';
+import './Grid.css';
+
+export default function Grid () {
+  const [state, dispatch] = useStateValue();
+  const [columns, setColumns] = useState(0);
+  const [rows, setRows] = useState(0);
+  const running = state.gameStatus.running;
+
+  const renderGrid = () => {
+    const removeHeight = 250;
+    const removeWidth = 20;
+    const columns =
+      Math.floor((window.innerWidth - removeWidth) / constants.CELL_SIZE[0]) -
+      1;
+    const rows =
+      Math.floor((window.innerHeight - removeHeight) / constants.CELL_SIZE[1]) -
+      1;
+    setColumns(
+      columns < constants.MIN_GRID_SIZE[0]
+        ? constants.MIN_GRID_SIZE[0]
+        : columns
+    );
+    setRows(
+      rows < constants.MIN_GRID_SIZE[1] ? constants.MIN_GRID_SIZE[1] : rows
+    );
+    dispatch({
+      type: 'SET_GRID',
+      payload: {
+        columns,
+        rows
+      }
+    });
+  };
+
+  const resizeHandler = useCallback(() => {
```

Comment:
there is no point to use `useCallback` here. Just create this function inside the `useEffect` to use the same reference for cleanup function

Reply:
nice recommendation again

refactored to 
```js
  useEffect(() => {
    const resizeHandler = () => {
      // block change while running
      if (running) {
        return
      }
      renderGrid()
    }

    renderGrid()
    window.addEventListener('resize', resizeHandler)
    return () => {
      window.removeEventListener('resize', resizeHandler)
    }
  }, [running])
```

I think is what you mean, just to be sure that there is not a misinterpretation